src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -1185,7 +1054,7 @@ private void DisposeConnections(List<IDisposable> instancesToDispose)
 
                     // ReSharper disable once PossibleUnintendedReferenceComparison
                     var handler = outgoingHandlers.Find(o => o.Destination == actual); // we explicitly compare references.
-                    if (handler == null)
+                    if (handler == null || handler is OutgoingReplicationHandlerBase outgoingHandler == false)
```

Comment:
`OutgoingReplicationHandlerBase` vs `AbstractOutgoingReplicationHandler` is _very_ confusing.... 
let's rename the `OutgoingReplicationHandlerBase`  to `DocumentDatabaseOutgoingReplicationHandler`

Reply:
and why is the condition above?